packages/affiliates/libs/affiliates/utils.js

```diff
@@ -362,7 +362,14 @@ function makeUnixPipe(cb, stdin = process.stdin) {
         return result + str;
       })
       // once string is final, we try to parse it as json. We assume that our caller is passing us a valid json string.
-      .map(x => JSON.parse(x))
+      .map(x => {
+        try {
+          return JSON.parse(x);
+        } catch (err) {
+          console.error(x);
```

Comment:
nit: why are we logging this error and rethrowing? Seems unrelated to the point of this PR and it also seems a bit strange to log the error that you're bubbling up? Won't it get printed twice?

Reply:
its actually not logging the error, its logging the object which triggered the parsing error. The reason is because the error only gives us a line within the object to refer to, but its useless without being able to actually see the object!